WordPress/wordpress-playground
# Blueprints: Add setSiteLanguage step to change the language

## Motivation for the change, related issues

[To unlock Playground to non-English speakers we want to make it easier for sites to be loaded in a local language.](https://github.com/WordPress/wordpress-playground/issues/1513)

Users can easily switch the site language by adding a `language=LANGUAGE_CODE` query parameter.

## Implementation details

This PR adds a `language` parameter to the Query API which adds a `setSiteLanguage` blueprint step.

We could have achieved this without a new step, but it would require adding two steps to the blueprint generator (add constant, run PHP).

The step downloads core, plugin, and theme translations from Download.WordPress.org and moves the files into the translation directory.

Step usage example:

```json
{
	"steps": [
		{
			"step": "setSiteLanguage",
			"language": "pl_PL"
		}
	]
}
```

## Testing Instructions (or ideally a Blueprint)

- Checkout this branch
- Open this [Playground url](http://127.0.0.1:5400/website-server/?language=es_ES&plugin=akismet&theme=abletone&php=8.0&wp=6.5&storage=none&networking=yes)
- Confirm that the theme is translated (the menu label should be _Menú_)
- Check the admin bar and confirm it's translated
- Open Akismet settings (/wp-admin/options-general.php?page=akismet-key-config&view=start) and confirm they are translated
